### DISCUSSION TITLE
Is multiple "Result webparts" really needed?

### DISCUSSION THREAD

**Discussion by pospik** [microsoft-search/pnp-modern-search#2722] [Q&A]
Hi,
I would like to use all combinations on my page (textbox + verticals + search result + filters). I need to have verticals with department names (ie IT, Business etc.) and after click on vertical I need to update results. I need to update results when user hit 'Enter' after finish writing to textbox. But I still need filtered results by vertical component. After researching I found, that I have to use many "Results" part for every vertical and hide them for any other vertical. I have 10 verticals and this is not too much confort for me, because when I change layout or something I have to change it in all results webparts. In case when I have only one results part connected to verticals and search box, it will not update after hitting "enter" in text box.. I hope that my description is not to much confusing... (sry) Can someone help me with this setup? It's for phone book in my case..
thx. Tomas

**Comment by wobba** [comment]
v4 uses one result wp per vertical. You can have the layout in a html file and load that for all wp's. This way you don't have to edit all wp's when changing the layout.
